Document the change in `Field.alias` behavior in Pydantic V2 (#6508)
pydantic/pydantic: docs/migration.md

`Field` no longer supports arbitrary keyword arguments to be added to the JSON schema. Instead, any extra
data you want to add to the JSON schema should be passed as a dictionary to the `json_schema_extra` keyword argument.

> In Pydantic V1, The `alias` property returns the field's name when no alias is set.

Reviewer: ```suggestion
In Pydantic V1, the `alias` property returns the field's name when no alias is set.
```

> We have changed this behavior in Pydantic V2 to return `None` when no alias is set.

Reviewer: ```suggestion
In Pydantic V2, this behavior has changed to return `None` when no alias is set.
```
